bokehjs/src/lib/models/widgets/numeric_spinner.ts

```diff
@@ -0,0 +1,173 @@
+  return 0
+}
+
+// Inpisration from https://github.com/uNmAnNeR/ispinjs
+export class NumericSpinnerView extends NumericInputView {
+  model: NumericSpinner
+
+  protected wrapper_el: HTMLDivElement
+  protected inc_el: HTMLButtonElement
+  protected dec_el: HTMLButtonElement
+  private _interval_handle: number;
+
+  *buttons(): Generator<HTMLButtonElement> {
+    yield this.inc_el
+    yield this.dec_el
+  }
+
+  initialize(): void {
+    super.initialize()
+  }
+
+  connect_signals(): void {
+    super.connect_signals()
+    const p = this.model.properties
+    this.on_change(p.disabled, () => {
+      for (const btn of this.buttons()) {
+        toggle_attribute(btn, "disabled", this.model.disabled)
+      }
+    })
+  }
+
+  render(): void {
+    super.render()
+    this.wrapper_el = div({class: "bk-spin-wrapper"})
+    this.group_el.replaceChild(this.wrapper_el, this.input_el)
+
+    this.inc_el = button({class: "bk-spin-btn bk-spin-btn-inc"})
+    this.dec_el = button({class: "bk-spin-btn bk-spin-btn-dec"})
+    this.wrapper_el.appendChild(this.input_el)
+    this.wrapper_el.appendChild(this.inc_el)
+    this.wrapper_el.appendChild(this.dec_el)
+    for (const btn of this.buttons()) {
+      toggle_attribute(btn, "disabled", this.model.disabled)
+      btn.addEventListener("mousedown", (evt) => this._btn_mouse_down(evt))
+      btn.addEventListener("mouseup", () => this._btn_mouse_up())
+      btn.addEventListener("mouseleave", () => this._btn_mouse_leave())
+    }
+    this.input_el.addEventListener("wheel", (evt) =>
+      this._input_mouse_wheel(evt)
+    )
+    this.input_el.addEventListener("keydown", (evt) =>
+      this._input_key_down(evt)
+    )
+  }
+
+  get precision(): number {
+    const {low, step} = this.model
+    return max(
+      ...[low, step]
+        .reduce<number[]>((prev, val) => {
+          if (val != null) prev.push(val)
+          return prev
+        }, [])
+        .map(precision)
+    )
+  }
+
+  _start_incrementation(sign: 1 | -1): void {
+    clearInterval(this._interval_handle)
+    const {step} = this.model
+    this._interval_handle = setInterval(() => this.increment(sign * step), 100)
+  }
+
+  _btn_mouse_down(evt: MouseEvent): void {
+    evt.preventDefault()
+    const sign = evt.currentTarget === this.inc_el ? 1 : -1
+    this.increment(sign * this.model.step)
+    this.input_el.focus()
+    //while mouse is down we increment at a certain rate
+    this._start_incrementation(sign)
+  }
+
+  _btn_mouse_up(): void {
+    clearInterval(this._interval_handle)
+  }
+
+  _btn_mouse_leave(): void {
+    clearInterval(this._interval_handle)
+  }
+
+  _input_mouse_wheel(evt: WheelEvent): void {
+    if (document.activeElement !== this.input_el) return
+    evt.preventDefault()
+
+    const sign = evt.deltaY > 0 ? -1 : 1
+    this.increment(sign * this.model.step)
+  }
+
+  _input_key_down(evt: KeyboardEvent) {
+    switch (evt.keyCode) {
+      case 38: // arrow up
```

Comment:
`core/dom` has `Keys` enum to help with this. If a value is missing, update the enum and use it here.